File: cyber/demo_cc/demo05_param_client.cc
#include "cyber/cyber.h"
#include "cyber/parameter/parameter_client.h"

using apollo::cyber::ParameterClient;
using apollo::cyber::Parameter;

int main(int argc, char const *argv[])
{
    apollo::cyber::Init(argv[0]);
    AINFO << "parameter client...";

    std::shared_ptr<apollo::cyber::Node> param_client_node = apollo::cyber::CreateNode("param_client");
    auto param_client = std::make_shared<ParameterClient>(param_client_node, "param");

    Parameter temp;
    param_client->GetParameter("type", &temp);
    AINFO << temp.Name() << " = " << temp.AsString();
    param_client->GetParameter("height", &temp);
    AINFO << temp.Name() << " = " << temp.AsDouble();
    param_client->GetParameter("lasers", &temp);
    AINFO << temp.Name() << " = " << temp.AsInt64();

    param_client->SetParameter(Parameter("lasers", 1));
    AINFO << "change .....";
    param_client->GetParameter("lasers", &temp);
    AINFO << temp.Name() << " = " << temp.AsInt64();


    apollo::cyber::WaitForShutdown();

    return 0;
}

File: cyber/demo_py/demo05_param_client.py
from cyber.python.cyber_py3 import cyber
from cyber.python.cyber_py3 import parameter

if __name__ == "__main__":
    cyber.init()
    print("param service ....")
    node = cyber.Node("camera_service")

    param_client = parameter.ParameterClient(node, "car_param")

    p1 = param_client.get_parameter("car_type")
    p2 = param_client.get_parameter("height")
    p3 = param_client.get_parameter("cameras")

    print(f"car_type = {p1.as_string()}, height = {p2.as_double()}, cameras = {p3.as_int64}")

    ps = param_client.get_paramslist()
    for p in ps:
        print(p.debug_string())

    cyber.shutdown()
